packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/findLastFinishingChildSpan.test.js

```diff
@@ -9,12 +9,16 @@ import sanitizeOverFlowingChildren from './sanitizeOverFlowingChildren';
 
 describe('findLastFinishingChildSpanId', () => {
   it('Should find lfc of a span correctly', () => {
-    const refinedSpanData = getChildOfSpans(test1.trace.spans);
-    const spanMap = refinedSpanData.reduce((map, span) => {
-      map.set(span.spanID, span);
-      return map;
-    }, new Map());
-    const sanitizedSpanMap = sanitizeOverFlowingChildren(spanMap);
+    const trace = test1.trace;
+    // Create spanMap with childSpanIds populated from childSpans
+    const spanMap = new Map();
+    trace.spans.forEach(span => {
+      const childSpanIds = span.childSpans ? span.childSpans.map(child => child.spanID) : [];
+      spanMap.set(span.spanID, { ...span, childSpanIds });
+    });
+
+    const refinedSpanMap = getChildOfSpans(spanMap);
+    const sanitizedSpanMap = sanitizeOverFlowingChildren(refinedSpanMap);
```

Comment:
instead of transforming the top map to contain childSpanIds pass the original map and change the function to work off that richer data.

IMPORTANT: make sure the function does not modify the top-level map. If it needs to modify it then it must make a copy.